lib/App/Sqitch/Command/check.pm

```diff
@@ -0,0 +1,208 @@
+package App::Sqitch::Command::check;
+
+use 5.010;
+use strict;
+use warnings;
+use utf8;
+use Locale::TextDomain qw(App-Sqitch);
+use App::Sqitch::X qw(hurl);
+use Moo;
+use App::Sqitch::Types qw(Str Target Engine Change);
+use Try::Tiny;
+use namespace::autoclean;
+
+extends 'App::Sqitch::Command';
+with 'App::Sqitch::Role::ContextCommand';
+with 'App::Sqitch::Role::ConnectingCommand';
+
+# VERSION
+
+has target_name => (
+    is  => 'ro',
+    isa => Str,
+);
+
+has target => (
+    is      => 'rw',
+    isa     => Target,
+    handles => [qw(engine plan plan_file)],
+);
+
+has date_format => (
+    is      => 'ro',
+    lazy    => 1,
+    isa     => Str,
+    default => sub {
+        shift->sqitch->config->get( key => 'status.date_format' ) || 'iso'
+    }
+);
+
+has project => (
+    is      => 'ro',
+    isa     => Str,
+    lazy    => 1,
+    default => sub {
+        my $self = shift;
+        try { $self->plan->project } catch {
+            # Just die on parse and I/O errors.
+            die $_ if try { $_->ident eq 'parse' || $_->ident eq 'io' };
+
+            # Try to extract a project name from the registry.
+            my $engine = $self->engine;
+            hurl status => __ 'Database not initialized for Sqitch'
+                unless $engine->initialized;
+            my @projs = $engine->registered_projects
+                or hurl status => __ 'No projects registered';
+            hurl status => __x(
+                'Use --project to select which project to query: {projects}',
+                projects => join __ ', ', @projs,
+            ) if @projs > 1;
+            return $projs[0];
+        };
+    },
+);
+
+sub options {
+    return qw(
+        project=s
+        target|t=s
+    );
+}
+
+sub execute {
+    my $self = shift;
+    my ($targets) = $self->parse_args(
+        target => $self->target_name,
+        args   => \@_,
+    );
+
+    # Warn on multiple targets.
+    my $target = shift @{ $targets };
+    $self->warn(__x(
+        'Too many targets specified; connecting to {target}',
+        target => $target->name,
+    )) if @{ $targets };
+
+    # Good to go.
+    $self->target($target);
+    my $engine = $target->engine;
```

Comment:
To follow the other database commands, I'd move most of the rest of this method to a `check` method in App::Sqitch::Engine. I think it's likely to share quite a bit of its logic with the `verify` method in that class, so it might make sense to factor out the common bits.